astropy/utils/state.py

```diff
@@ -7,7 +7,79 @@
 __all__ = ['ScienceState']
 
 
-class ScienceState:
+class StateMixinMeta(type):
```

Comment:
I'm very uncertain about this new meta class (see general comments). If we do it at all, my tendency would be to expose not a `__getitem__` but rather a `__getattr__`.

Reply:
I agree about the  __getattr__ over __getitem__. I have resolved the issue in another branch. My general comment (below) will address this and the meta class.